src/tools/tidy/src/cargo_lock.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::path::Path;
+
+fn filter_lock_files(path: &Path) -> bool {
+    use std::fs;
+
+    match fs::metadata(path) {
+        Ok(m) => {
+            if !m.is_dir() {
+                return true;
+            }
+            let dir = fs::read_dir(path);
+            for entry in t!(dir).map(|e| t!(e)) {
+                if entry.file_name().to_str() == Some("Cargo.lock") {
+                    return false;
+                }
+            }
+            return true;
+        },
+        Err(_) => { true }
+    }
+}
```

Comment:
The filter may not be the most applicable here because in theory we want to recurse into all directories in the tree?